florist/api/launch.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import sys
+import time
+from multiprocessing import Process
+from typing import Callable, Sequence
+
+import flwr as fl
+from fl4health.clients.basic_client import BasicClient
+from fl4health.server.base_server import FlServer
+from flwr.common.logger import DEFAULT_FORMATTER
+from flwr.server import ServerConfig
+
+
+def redirect_logging_from_console_to_file(log_file_path: str) -> None:
+    """
+    Function that redirects loggers outputing to console to specified file.
+
+    Args:
+        log_file_path (str): The path to the file to log to.
+    """
+
+    # Define file handler to log to and set format
+    fh = logging.FileHandler(log_file_path)
+    fh.setFormatter(DEFAULT_FORMATTER)
+
+    # Loop through existing loggers to check if they have one or more streamhandlers
+    # If they do, remove them (to prevent logging to the console) and add filehandler
+    for name in logging.root.manager.loggerDict:
+        logger = logging.getLogger(name)
+        if not all([isinstance(h, logging.StreamHandler) is False for h in logger.handlers]):
+            logger.handlers = [h for h in logger.handlers if not isinstance(h, logging.StreamHandler)]
+            logger.addHandler(fh)
+
+
+def start_server(
+    server_constructor: Callable[..., FlServer],
+    server_address: str,
+    n_server_rounds: int,
+    server_log_file_name: str,
+) -> None:
+    """
+    Function to start server that executes inside process spawned by launch_server.
+    Redirects logging to console, stdout and stderr to file.
+
+    Args:
+        server_constructor (Callable[FlServer]): Callable that constructs FL server.
+        server_address (str): String of <IP>:<PORT> to make server available.
+        n_server_rounds (str): The number of rounds to perform FL
+    """
+    redirect_logging_from_console_to_file(server_log_file_name)
+    log_file = open(server_log_file_name, "a")
+    # Send remaining ouput (ie print) from stdout and stderr to file
+    sys.stdout = sys.stderr = log_file
+    server = server_constructor()
+    fl.server.start_server(
+        server=server,
+        server_address=server_address,
+        config=ServerConfig(num_rounds=n_server_rounds),
+    )
+    server.metrics_reporter.dump()
+    server.shutdown()
```

Comment:
Put the `shutdown` before `dump` in case there is something being written to the metrics by the `shutdown` function.